src/services/selectPerson.ts

```diff
@@ -0,0 +1,14 @@
+import { SetStateAction } from 'react';
+import { Person } from '../types';
+
+export function selectPerson(
+  people: Person[],
+  setSelectedPerson: React.Dispatch<SetStateAction<Person | null>>,
+  currentParent: Person,
+) {
+  const selectedParent = people.find(p => p.name === currentParent.name);
```

Comment:
The function uses `name` for comparison, which might not be unique. Consider using a unique identifier like `slug` for more reliable comparisons.